### PR TITLE
Add API conversion interface for Dropout layer

### PR DESCRIPTION
**Pull request by abhaikollara**

### CODE REVIEW

File: keras/legacy/interfaces.py [hunk 1]
```diff
@@ -74,3 +74,62 @@ def wrapper(*args, **kwargs):
...
+            if 'rate' in kwargs:
+                raise_duplicate_arg_error('p', 'rate')
+            rate = kwargs.pop('p')
+            args = (args[0], rate)
+
+        converted = []
+
+        # Remaining kwargs.
+        conversions = [
+            ('p', 'rate'),
```
Comment (jihobak) [inline keras/legacy/interfaces.py]: 'p' is already removed when, kwargs.pop('p')

Comment (abhaikollara) [inline keras/legacy/interfaces.py]: Thanks for pointing out, will fix it

File: keras/legacy/interfaces.py [hunk 2]
```diff
@@ -74,3 +74,36 @@ def wrapper(*args, **kwargs):
...
+            if len(args) > 1:
+                raise TypeError('Got both a positional argument '
+                                'and keyword argument for argument '
+                                '`rate` '
+                                '(`p` in the legacy interface).')
+            if 'rate' in kwargs:
+                raise_duplicate_arg_error('p', 'rate')
+            rate = kwargs.pop('p')
+            args = (args[0], rate)
+            warnings.warn('Update your `Dropout` layer call to Keras 2 API. Arguement `p` has been replaced by `units`')
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: This should contain the copy-and-pastable code of the call layer (like we do for the Dense layer).

File: tests/keras/legacy/interface_test.py [hunk 3]
```diff
@@ -28,5 +28,11 @@ def test_dense_legacy_interface():
     assert json.dumps(old_layer.get_config()) == json.dumps(new_layer.get_config())
 
 
+@keras_test
+def test_dropout_legacy_interface():
+    old_layer = keras.layers.Dropout(p=3, name='drop')
```
Comment (fchollet) [inline tests/keras/legacy/interface_test.py]: Please also add a test checking `rate` as a positional argument (rather than keyword argument).

File: keras/legacy/interfaces.py (hunk 2 above)
Comment (abhaikollara) [inline keras/legacy/interfaces.py]: Should the copy-pastable code have keyword arguments like `Dropout(rate=0.5)` or the one with positional arguements `Dropout(0.5)`

Comment (fchollet) [inline keras/legacy/interfaces.py]: It should contain the same thing as what the user wrote, but converted to the new API.

File: tests/keras/legacy/interface_test.py (hunk 3 above)
Comment (abhaikollara) [inline tests/keras/legacy/interface_test.py]: @fchollet I've added another test case, I hope this is what you intended

File: keras/legacy/interfaces.py [hunk 4]
```diff
@@ -74,3 +74,37 @@ def wrapper(*args, **kwargs):
...
+            if len(args) > 1:
+                raise TypeError('Got both a positional argument '
+                                'and keyword argument for argument '
+                                '`rate` '
+                                '(`p` in the legacy interface).')
+            if 'rate' in kwargs:
+                raise_duplicate_arg_error('p', 'rate')
+            rate = kwargs.pop('p')
+            args = (args[0], rate)
+            signature = '`Dropout(rate=`' + str(args[1]) + ')`'
```
Comment (fchollet) [inline keras/legacy/interfaces.py]: This ignores additional keyword arguments possibly passed by the user.
